src/tests/Microsoft.PowerFx.Interpreter.Tests/ConfigTests.cs

```diff
@@ -370,98 +371,184 @@ public void RecalcEngine_Symbol_CultureInfo4()
-                    Assert.Equal("ındıgo indigo", (result as StringValue).Value);
+            result = engine2.Eval(properExpression);
+            Assert.Equal("Indıgo İndigo", (result as StringValue).Value);
+        }
 
-                    result = engine2.Eval(properExpression);
-                    Assert.Equal("Indıgo İndigo", (result as StringValue).Value);
-                }
-                catch (Exception ex)
-                {
-                    exception = ex;
-                }
-            });
+        // Verify if text is transformed using the correct culture info (PowerFxConfig and global settings)
+        // Origin: https://github.com/microsoft/Power-Fx/issues/111
+        [Fact]
+        public void RecalcEngine_Symbol_CultureInfo6()
+        {
+            RunOnIsolatedThread(new CultureInfo("bg-BG"), RecalcEngine_Symbol_CultureInfo6_ThreadProc);
+        }
 
-            t.Start();
-            t.Join();
+        private void RecalcEngine_Symbol_CultureInfo6_ThreadProc()
+        {
+            const string formula = "Concatenate(\"Hello\", \" World!\")";
+            var defaultCulture = CultureInfo.CreateSpecificCulture("en");
+            var engine = new RecalcEngine(new PowerFxConfig(defaultCulture));
+            var result = engine.Eval(formula);
 
-            if (exception != null)
+            var helloWorld = Assert.IsType<string>(result.ToObject());
+            Assert.Equal("Hello World!", helloWorld);
+        }
+
+        // Verify BombCulture object indeed fails when accessed.
+        [Fact]
+        public void TestBombCulture()
+        {
+            // Ignore thread's current culture. 
+            RunOnIsolatedThread(new BombCulture(), () =>
             {
-                throw exception;
-            }
+                var engine = new Engine();
+                var check = new CheckResult(engine);
+                check.SetText("1+2"); // will pull from CurrentThread. 
+
+                // Verify BombCulture works.
+                Assert.Throws<NotImplementedException>(() => check.ApplyParse());
+            });
         }
 
-        // Verify if text is transformed using the correct culture info (PowerFxConfig and global settings)
-        // Origin: https://github.com/microsoft/Power-Fx/issues/111
+        private static readonly CultureInfo _bombCulture = new BombCulture();
+
+        // Test explicitly setting cultures at each level:
+        // - parser 
+        // - compile-time errors 
+        // - runtime evaluation 
+        // These ignore the Threads's / Config / Engine's current culture.
         [Fact]
-        public void RecalcEngine_Symbol_CultureInfo6()
+        private void MultiCulture1()
         {
-            Exception exception = null;
+            RunOnIsolatedThread(_bombCulture, MultiCulture1ThreadProc);
+        }
 
-            var t = new Thread(() =>
+        // Test Parse and errors 
+        private void MultiCulture1ThreadProc()
```

Comment:
These new MultiCulture* tests are the key addition here. 
The rest of the changes are noisy diffs, due to refactoring to use RunOnIsolatedThread helper